onecodex/auth.py

```diff
@@ -171,22 +171,40 @@ def _logout(creds_file=None):
 
 
 def login_required(fn):
-    """
-    Decorator for the CLI for requiring login before proceeding.
+    """Requires login before proceeding, but does not prompt the user to login. Decorator should
+    be used only on Click CLI commands.
+
+    Notes
+    -----
+    Different means of authentication will be attempted in this order:
+        1. An API key present in the Click context object from a previous successful authentication.
+        2. A bearer token (ONE_CODEX_BEARER_TOKEN) in the environment.
+        3. An API key (ONE_CODEX_API_KEY) in the environment.
+        4. An API key in the credentials file (~/.onecodex).
     """
 
     @wraps(fn)
     def login_wrapper(ctx, *args, **kwargs):
-        if 'API_KEY' in ctx.obj and ctx.obj['API_KEY'] is not None:
-            ctx.obj['API'] = Api(api_key=ctx.obj['API_KEY'], telemetry=ctx.obj['TELEMETRY'])
+        api_kwargs = {'telemetry': ctx.obj['TELEMETRY']}
```

Comment:
Looks good but I think we should document the precedence that is used to set the api key b/c I've had issues with packages before where I could not figure out how the API key was determined.